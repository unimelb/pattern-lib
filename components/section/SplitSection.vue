<template>
  <section
    :class="{ 'split-section--withquote': quote }"
    class="split-section">
    <div
      :class="{ 'split-section__side--with-image': imageLeft }"
      :style="{ backgroundImage: imageLeft ? `url(${bgImage})` : '' }"
      class="split-section__side">
      <SplitSectionQuote
        v-if="quote && imageLeft"
        :quote="quote"
        :cite="cite" />
      <div
        v-if="imageRight"
        class="split-section__inner">
        <slot>default left text</slot>
      </div>
    </div>
    <div
      :class="{ 'split-section__side--with-image': imageRight }"
      :style="{ backgroundImage: imageRight ? `url(${bgImage})` : '' }"
      class="split-section__side">
      <SplitSectionQuote
        v-if="quote && imageRight"
        :quote="quote"
        :cite="cite" />
      <div
        v-if="imageLeft"
        class="split-section__inner">
        <slot>default right text</slot>
      </div>
    </div>
  </section>
</template>

<script>
import SplitSectionQuote from './SplitSectionQuote.vue';

export default {
  name: 'SplitSection',
  components: { SplitSectionQuote },
  props: {
    imageLeft: {
      type: Boolean,
      default: false,
    },
    imageRight: {
      type: Boolean,
      default: false,
    },
    bgImage: {
      type: String,
      default: '',
    },
    quote: {
      type: String,
      default: '',
    },
    cite: {
      type: String,
      default: '',
    },
  },
};
</script>
