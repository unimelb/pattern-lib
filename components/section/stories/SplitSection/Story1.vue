<template>
  <main class="bg-white">
    <SplitSection
      image-left
      bg-image="https://picsum.photos/seed/1/1200/540">
      <h3 class="heading-section">
        This is a header
      </h3>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nesciunt error architecto provident beatae autem optio, et vel repellendus assumenda aliquid, non cumque esse aliquam neque earum voluptatibus, perferendis maiores obcaecati.
      </p>
      <a
        href="https://www.unimelb.edu.au"
        class="btn btn--icon btn--icon--chevron-right">
        Call to Action
      </a>
    </SplitSection>
  </main>
</template>

<script>
import SplitSection from '../../SplitSection.vue';

export default {
  components: { SplitSection },
  readme: { html: true, source: false },
};
</script>
