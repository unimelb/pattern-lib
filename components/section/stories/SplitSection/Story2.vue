<template>
  <main class="bg-white">
    <SplitSection
      image-right
      bg-image="https://picsum.photos/seed/1/1200/540">
      <h3 class="heading-section">
        This is a header
      </h3>
      <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nesciunt error architecto provident beatae autem optio, et vel repellendus assumenda aliquid, non cumque esse aliquam neque earum voluptatibus, perferendis maiores obcaecati.</p>
    </SplitSection>
  </main>
</template>

<script>
import SplitSection from '../../SplitSection.vue';

export default {
  components: { SplitSection },
  readme: { source: true, html: true },
};
</script>
