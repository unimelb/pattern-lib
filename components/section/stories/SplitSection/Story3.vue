<template>
  <main class="bg-white">
    <SplitSection
      image-left
      quote="University of Melbourne has provided opportunities i could never imagine. This is a ridiculously long quote because people like to be especially verbose. eos minima at"
      cite="Rebecca Lam, Molecular Biology"
      bg-image="https://picsum.photos/seed/1/1200/600">
      <h3 class="heading-section">
        This is a header
      </h3>
      <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nesciunt error architecto provident beatae autem optio, et vel repellendus assumenda aliquid, non cumque esse aliquam neque earum voluptatibus, perferendis maiores obcaecati.</p>
    </SplitSection>
  </main>
</template>

<script>
import SplitSection from '../../SplitSection.vue';

export default {
  components: { SplitSection },
  readme: { source: true, html: true },
};
</script>
