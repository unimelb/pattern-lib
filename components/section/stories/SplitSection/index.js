import { storiesOf } from '@storybook/vue';
import { createStory } from 'storybook/utils';

import Story1 from './Story1.vue';
import Story2 from './Story2.vue';
import Story3 from './Story3.vue';
import Story4 from './Story4.vue';
/* ##Import story component here */

/* Section - Focus  */
storiesOf('Section/Split Section', module)
  .add('Left', createStory(Story1))
  .add('Right', createStory(Story2))
  .add('Quote Left', createStory(Story3))
  .add('Quote Right', createStory(Story4));
/* ##Story goes here */
