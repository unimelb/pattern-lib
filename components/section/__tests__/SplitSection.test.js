import {
  shallow,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import SplitSection from '../SplitSection.vue';

expect.extend(toHaveNoViolations);

describe('SplitSection', () => {
  it('should match snapshot', () => {
    const result = shallow(SplitSection).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(SplitSection).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
