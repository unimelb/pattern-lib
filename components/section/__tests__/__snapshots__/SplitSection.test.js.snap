// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplitSection should match snapshot 1`] = `
<section
  class="split-section"
>
  <div
    class="split-section__side"
  >
    <!---->
     
    <!---->
  </div>
   
  <div
    class="split-section__side"
  >
    <!---->
     
    <!---->
  </div>
</section>
`;
